### PR TITLE
Feature/Dialogue submit on pressing Enter key

### LINKED ISSUE #154: Enter key can't be used to submit forms
🐞 **Problem**
  Currently in modals like creating board and columns, enter key can't be used to submit when focus 
  is in input fields. Users will have to press tap to focus on `Add`/`Create` or click with mouse.
 
🎯 **Goal**
It is a standard pattern to submit a form by clicking enter key in input field. We wanna do that

💡 **Possible solutions**
On press of enter/return key in input field, submit form

📋 **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - Addresses #<put issue number here> 
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by alanjsph**
Fixes #154 

Please Review

### CODE REVIEW

**Comment by Roshanjossey** [comment]
Hey @alanjsph, it might be good to extract

```
($event.keyCode == 13)?addNewColumn(newColumn):null
```

out to a function. One reason is that you're using it 3 times. Another better reason is that it's better to give a name to that action.

How about something like

```
$scope.submitForm = function(event) {
  (event.keyCode == 13)?addNewColumn(newColumn):null
}
```

and call it like

```
ng-keyup="submitForm(e)"
```

**Comment by alanjsph** [comment]
Sure, Ill work on it

**Comment by alanjsph** [comment]
How about this commit? @Roshanjossey

**Comment by Roshanjossey** [comment]
@alanjsph: Looks good. We'll wait for @glauberramos to have a look at this.

**Comment by alanjsph** [comment]
@Roshanjossey  Sounds good.

**Comment by glauberramos** [comment]
Hey guys,

I just tested the PR it looks good. But I have a couple of comments:

- We need to add some unit tests to test this new method.
- Also why are we keeping the console.log into the code? Production code should not have console.log since it is not doing anything actually.

File: js/mainController.js [hunk 1]
```diff
@@ -235,6 +235,30 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(e, method, data){
```
Comment (glauberramos) [inline js/mainController.js]: We should use variable name as "event", like we are using on other parts of the code.

File: js/mainController.js [hunk 2]
```diff
@@ -235,6 +235,30 @@ angular
...
+                  console.log("Invalid board name");
+                }
+                break;
+            case "addNewColumn":
+                if(data){
+                  $scope.addNewColumn(data);
+                  $scope.newColumn='';
+                }
+                else{
+                  console.log("Invalid column name")
```
Comment (glauberramos) [inline js/mainController.js]: Same here

File: js/mainController.js [hunk 3]
```diff
@@ -235,6 +235,30 @@ angular
...
+                  $scope.createNewBoard()
+                }
+                else{
+                  console.log("Invalid board name");
+                }
+                break;
+            case "addNewColumn":
+                if(data){
+                  $scope.addNewColumn(data);
+                  $scope.newColumn='';
```
Comment (glauberramos) [inline js/mainController.js]: Equal sign should have spaces around it in order to have better readability.

File: components/dialogs.html [hunk 4]
```diff
@@ -13,16 +13,17 @@
...
-  <input ng-model="newColumn" type="text"></input>
+  <input ng-model="newColumn" type="text" ng-keyup="submitOnEnter($event, 'addNewColumn', newColumn)"></input>
   <button ng-click="addNewColumn(newColumn); newColumn='';">Add</button>
 </script>
 <script type="text/ng-template" id="addNewBoard">
   <h1>New board</h1>
+
   <label>Board name</label>
-  <input ng-model="newBoard.name" type="text"></input>
+  <input ng-model="newBoard.name" type="text" ng-keyup="submitOnEnter($event, 'createNewBoard', 'null')"></input>
```
Comment (glauberramos) [inline components/dialogs.html]: I think you don't need to pass the third parameter here. If you pass only 2 it should work the same.

File: js/mainController.js [hunk 5]
```diff
@@ -235,6 +235,30 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(e, method, data){
+        if(e.keyCode == 13){
```
Comment (glauberramos) [inline js/mainController.js]: If's and else's should have space after the word, instead of "if(..){", should be "if (..) {"

**Comment by alanjsph** [comment]
All right I'll fix them all

**Comment by alanjsph** [comment]
@Roshanjossey @glauberramos  please review the code and see if the tests are good enough.

File: components/dialogs.html [hunk 6]
```diff
@@ -13,16 +13,17 @@
...
 <script type="text/ng-template" id="addNewBoard">
   <h1>New board</h1>
+
   <label>Board name</label>
-  <input ng-model="newBoard.name" type="text"></input>
+  <input ng-model="newBoard.name" type="text" ng-keyup="submitOnEnter($event, 'createNewBoard')"></input>
   <label>Max votes per user <small>(whole board)</small></label>
-  <input ng-model="newBoard.max_votes" type="number" ng-init="newBoard.max_votes = 6"></input>
-  <button ng-disabled="isBoardNameInvalid()"  ng-click="createNewBoard()">Create</button>
+  <input ng-model="newBoard.max_votes" type="number" ng-init="newBoard.max_votes = 6" ng-keyup="submitOnEnter($event, 'createNewBoard', 'null')"></input>
```
Comment (glauberramos) [inline components/dialogs.html]: You can remove 'null' from here

File: js/mainController.js [hunk 7]
```diff
@@ -235,6 +235,24 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(event, method, data){
+        if(event.keyCode == 13){
```
Comment (glauberramos) [inline js/mainController.js]: Add spaces here

**Comment by alanjsph** [comment]
@glauberramos I've removed the spaces for IF and removed null. please review the new commit

**Comment by glauberramos** [comment]
jsHint is breaking on travis:

js/mainController.js: line 239, col 29, Expected '===' and instead saw '=='.
js/mainController.js: line 241, col 34, Strings must use singlequote.
js/mainController.js: line 243, col 42, Missing semicolon.
js/mainController.js: line 246, col 32, Strings must use singlequote.

File: js/mainController.js [hunk 8]
```diff
@@ -235,6 +235,24 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(event, method, data){
+        if (event.keyCode == 13) {
+          switch (method){
+            case "createNewBoard":
+                if (!$scope.isBoardNameInvalid()) {
+                  $scope.createNewBoard()
```
Comment (glauberramos) [inline js/mainController.js]: missing ;

File: js/mainController.js [hunk 9]
```diff
@@ -235,6 +235,24 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(event, method, data){
+        if (event.keyCode == 13) {
+          switch (method){
+            case "createNewBoard":
```
Comment (glauberramos) [inline js/mainController.js]: use single quote

File: js/mainController.js [hunk 10]
```diff
@@ -235,6 +235,24 @@ angular
...
+                if (!$scope.isBoardNameInvalid()) {
+                  $scope.createNewBoard()
+                }
+                break;
+            case "addNewColumn":
```
Comment (glauberramos) [inline js/mainController.js]: use single quote

File: js/mainController.js [hunk 11]
```diff
@@ -235,6 +235,24 @@ angular
         } else return '';
       };
 
+      $scope.submitOnEnter = function(event, method, data){
+        if (event.keyCode == 13) {
```
Comment (glauberramos) [inline js/mainController.js]: use ===

**Comment by alanjsph** [comment]
All done. @glauberramos

**Comment by alanjsph** [comment]
@glauberramos When would this feature reflect in the app running in github.io?

**Comment by glauberramos** [comment]
@alanjsph very soon, I must need to do some tests and I will deploy it. Probably in the end of the day today (brazil time)

**Comment by glauberramos** [comment]
It is on production

**Comment by alanjsph** [comment]
Yay! My first contribution. Thank you @Roshanjossey @glauberramos

**Pull request merged**